document-generator-core/src/test/java/uk/gov/companieshouse/document/generator/core/controller/DocumentGeneratorControllerTest.java

```diff
@@ -0,0 +1,156 @@
+package uk.gov.companieshouse.document.generator.core.controller;
+
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import uk.gov.companieshouse.document.generator.core.models.DocumentRequest;
+import uk.gov.companieshouse.document.generator.core.models.DocumentResponse;
+import uk.gov.companieshouse.document.generator.core.service.DocumentGeneratorService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DocumentGeneratorControllerTest {
+
+    @Mock
+    private BindingResult mockBindingResult;
+
+    @Mock
+    private DocumentGeneratorService mockDocumentGeneratorService;
+
+    @InjectMocks
+    private DocumentGeneratorController documentGeneratorController;
+
+    @Test
+    @DisplayName("Tests if Bad Request error return when binding error found")
+    public void errorReturnAsIncompleteRequest() {
+
+        DocumentRequest request = new DocumentRequest();
+
+        when(mockBindingResult.hasErrors()).thenReturn(true);
+
+        ResponseEntity responseEntity = documentGeneratorController.generateDocument(request, mockBindingResult);
+
+        assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("Test if Created returned if response not null and location present")
+    public void createdReturned() {
+
+        DocumentRequest request = setDocumentgeneratorRequest();
+
+        DocumentResponse response = new DocumentResponse();
+        response.setDescription("description");
```

Comment:
These variables are used more than once and could be made `static final` on the class